stress_test/controller.py

```diff
@@ -0,0 +1,402 @@
+
+        self.status = 'BUILT'
+
+    def wait_until_listens(interval_ms):
+        """ Waits for controller to start listening on specified port.
+
+        :param interval_ms: milliseconds to wait (in milliseconds).
+        :raises Exception: If controller fails to start or if another process
+        listens on controllers port.
+        :rtype int
+        :type interval_ms: int
+        """
+
+        logging.info('[Controller] Waiting to start listening on a port')
+
+        timeout = time.time() + (float(interval_ms) / 1000)
+        while time.time() < timeout:
+            time.sleep(1)
+            gpid = util.process.getpid_listeningonport(self.port, self._init_ssh)
+            logging.info('Returned pid listening on port {0}: {1}'.
+                          format(self.port, gpid))
+            if gpid == 0:
+                raise Exception('Another controller seems to have started in the '
+                                'meantime. Exiting...')
+
+        raise Exception('Controller failed to start within a period of {0} '
+                        'minutes'.format(timeout))
+
+
+    def wait_until_up(interval_ms):
+        """ Waits for controller status to become 1 (started).
+
+        :param interval_ms: milliseconds to wait (in milliseconds).
+        :raises Exception: If controller fails to start or if another process
+        listens on controllers port.
+        :type interval_ms: int
+        """
+
+        logging.info('[Controller] Waiting to be started')
+
+        timeout = time.time() + (float(interval_ms) / 1000)
+        while time.time() < timeout:
+            time.sleep(1)
+            if check_status() == '1':
+                return
+
+        raise Exception('Controller failed to start. '
+                        'Status check returned 0 after trying for {0} seconds.'.
+                        format(float(interval_ms) / 1000))
+
+
+
+class ODL(Controller):
+
+    def __init__(self, ctrl_base_dir, test_config):
+
+        super(self.__class__, self).__init__(ctrl_base_dir, test_config)
+
+        self.stat_period_ms = test_config['controller_statistics_period_ms']
+
+        if 'controller_flowmods_conf_handler' in test_config:
+            self.flowmods_conf_hnd= ctrl_base_dir + test_config['controller_flowmods_conf_handler']
+
+        if 'controller_statistics_handler' in test_config:
+            self.statistics_hnd = ctrl_base_dir + test_config['controller_statistics_handler']
+
+        if 'controller_persistent_handler' in test_config:
+            self.persistent_hnd = ctrl_base_dir + test_config['controller_persistent_handler']
+
+        if 'controller_restconf_port' in test_config:
+            self.restconf_port = test_config['controller_restconf_port']
+            self.restconf_user = test_config['controller_restconf_user']
+            self.restconf_password = test_config['controller_restconf_password']
+
+        self.hosts = -1
```

Comment:
good addition! 

better rename these attributes to `oper_hosts`, `oper_switches`, etc., to denote that these are numbers acquired from the ODL operational datastore